Fix bug in Table#query which caused it to return a row for an id that…

ahutchings opened the pull request:
… does not exist in the current result set

Refs #99

plandem commented:
@tommikaikkonen that's why I postponed merge request, because of full scan. There is also request to add **custom indexes** (not only by ID like now) - it would better to have same solution, but not sure how to implement it yet. Need time to think.

ahutchings commented:
Updated the PR to restore the lookup from the index by id using the strategy proposed by @tommikaikkonen of sorting the clauses to have the id filter queries first. I also added the clause `index === 0` check to the index lookup because it is not safe to do unless the id query is first.

ahutchings commented:
Updated the truthy check in the id lookup block to a not-null & not-undefined check as described. I think this is ready to go pending further review.

plandem commented:
@tommikaikkonen so publish a new version or you will do it by self?

tommikaikkonen commented:
Would be great if you can do it @plandem. If not, I have a bit of time tomorrow to get it out.

plandem commented:
@tommikaikkonen published + updated changes

ahutchings commented:
Great, thanks all 👍

plandem commented:
@ahutchings nope, thanks to you :)

markerikson commented:
@tommikaikkonen , @plandem : I see that the NPM package has been updated, but there's no corresponding releases since 0.9.0-rc3 in the Github releases tab.  Think we could get that updated too?

plandem commented:
@markerikson probably I published modules by differ way :( **npm run pub** from our **package.json** I'm not pro in this area

markerikson commented:
Yeah, I've never done an NPM release _or_ a Github release for anything.  I've got a project that I'm updating from 0.8.x to 0.9.x, was about to update redux-orm using Yarn, and realized I wasn't sure what the current version was.  Checked the Github releases page first and saw nothing new, then checked the NPM page and saw we are indeed at 0.9.4.  So, basically hoping we can keep anyone else from being confused.

nealoke commented:
@markerikson yep totally agree, most clear is to add it to the releases tab 👍
